internal/provider/resource_directory_role_collection_assignment.go

```diff
@@ -96,6 +98,29 @@ func (rs *directoryRoleCollectionAssignmentResource) Schema(_ context.Context, _
 				PlanModifiers: []planmodifier.String{
 					stringplanmodifier.RequiresReplace(),
 				},
+				Validators: []validator.String{
+					stringvalidator.LengthAtLeast(1),
+					stringvalidator.AlsoRequires(path.MatchRoot("origin")),
+				},
+			},
+			"attribute_name": schema.StringAttribute{
+				MarkdownDescription: "The name of the attribute to assign.",
+				Optional:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.AlsoRequires(path.MatchRoot("attribute_value")),
+					stringvalidator.AlsoRequires(path.MatchRoot("origin")),
+					stringvalidator.LengthAtLeast(1),
+				},
+			},
+			"attribute_value": schema.StringAttribute{
+				MarkdownDescription: "The value of the attribute to assign.",
+				Optional:            true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
```

Comment:
I expected an additional modifier:

stringvalidator.AlsoRequires(path.MatchRoot("attribute_name")),

Otherwise it is possible to provide an attribute value together with a user or a group, correct?

Reply:
You are right. Added this validation